### PR TITLE
Add create collection api endpoint

### PR DESCRIPTION
**Pull request by Aaronel10**

### CODE REVIEW

File: web/src/pages/api/addPostToCollection.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+  // get existing posts
+
+  const resStatus = await prisma.collection.update({
+    where: {
+      id: String(collectionId),
+    },
+    data: {
+      posts: {
+        connect: { id: postToAdd.id },
```
Comment (Sleepingpirates) [inline web/src/pages/api/addPostToCollection.ts:44]: Good job with the connect here

**Review by Sleepingpirates — APPROVED** [review]

File: web/src/pages/api/addPostToCollection.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (!postId) {
+    res.statusCode = 400;
+    return res.json("Give me a post for the collection");
+  }
+
+  // find the post
+  const postToAdd = await prisma.post.findFirst({
+    where: {
+      id: String(postId),
```
Comment (rob-3) [inline web/src/pages/api/addPostToCollection.ts:28]: I would suggest `postId.toString()`, which will throw if postId is null or undefined somehow

File: web/src/pages/api/addPostToCollection.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      id: String(collectionId),
+    },
+    data: {
+      posts: {
+        connect: { id: postToAdd.id },
+      },
+    },
+  });
+
+  return res.json({ sucess: true });
```
Comment (rob-3) [inline web/src/pages/api/addPostToCollection.ts:49]: `success`

File: web/src/pages/api/addPostToCollection.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (!postToAdd) {
+    res.statusCode = 400;
+    return res.json("Invalid PostID");
+  }
+  // get existing posts
+
+  const resStatus = await prisma.collection.update({
+    where: {
+      id: String(collectionId),
```
Comment (rob-3) [inline web/src/pages/api/addPostToCollection.ts:40]: Same as above

File: web/src/pages/api/createCollection.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  const query = req.query;
+
+  if (!query.name) {
+    res.statusCode = 400;
+    return res.json("Give me a name for the collection");
+  }
+
+  const collection = await prisma.collection.create({
+    data: {
+      // userId: String(session.user?.id),
```
Comment (rob-3) [inline web/src/pages/api/createCollection.ts:27]: as above

File: web/src/pages/api/createCollection.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      userId: "cl9oql88i0009yb3z8wr3gzgy",
+      name: String(query.name),
```
Comment (rob-3) [inline web/src/pages/api/createCollection.ts:29]: as above

File: web/src/pages/api/removePostFromCollection.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      id: String(collectionId),
+    },
+    data: {
+      posts: {
+        disconnect: { id: postToAdd.id },
+      },
+    },
+  });
+
+  return res.json({ sucess: true });
```
Comment (rob-3) [inline web/src/pages/api/removePostFromCollection.ts:49]: `success`

**Review by rob-3 — CHANGES_REQUESTED** [review]
Just a few minor changes - I also am not certain, but I think that you don't have to return in Next.js, ie I think just `res.json()` is sufficient instead of `return res.json`

**Comment by rob-3** [comment]
There's a couple of minor tweaks I'm gonna make, but they're pretty minor so I'll just add them and get this merged tonight. Good work

**Comment by rob-3** [comment]
Eh, it's fine actually - approved

**Review by rob-3 — APPROVED** [review]

**Pull request merged**